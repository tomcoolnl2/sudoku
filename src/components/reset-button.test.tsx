
import { render, screen } from '@testing-library/react'
import { withTheme } from '../test'
import { ResetGameButton, ResetGameButtonProps } from './'


const props: ResetGameButtonProps = {
	reset: jest.fn()
}

describe('ResetGameButton', () => {
	
	it('should match it\'s snapshot', () => {
		const { container } = render(withTheme(<ResetGameButton {...props} />))
		expect(container.firstChild).toMatchSnapshot()
	})

	it('should be present within the DOM', () => {
		render(withTheme(<ResetGameButton {...props} />))
		const buttonElement = screen.getByText(/New Game/i)
		expect(buttonElement).toBeInTheDocument()
	})
})