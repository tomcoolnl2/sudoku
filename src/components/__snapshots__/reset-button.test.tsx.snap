// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetGameButton should match it's snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #282c34;
  border: 2px solid #282c34;
  border-radius: 4px;
  color: white;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 16px;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 4px 2px;
  min-height: 40px;
  padding: 0;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  opacity: .9;
}

.c0:disabled {
  opacity: .7;
  cursor: default;
}

.c0:focus {
  border-color: #EDDA74;
  outline: none;
}

.c0:hover {
  opacity: .7;
}

.c0 > svg {
  margin-right: 10px;
  fill: white;
  position: relative;
  width: 24px;
  height: 24px;
}

<button
  class="c0"
  data-testid="reset-button"
>
  <svg>
    reload.svg
  </svg>
   New Game
</button>
`;
